ateam_kenobi/src/path_planning/path_planner.cpp

```diff
@@ -0,0 +1,233 @@
+  const auto step_vector = direction_vector * (options.collision_check_resolution / segment_length);
+  const int step_count = segment_length / options.collision_check_resolution;
+  for (auto step = 0; step < step_count; ++step) {
+    const auto state = p1 + (step * step_vector);
+    if (!isStateValid(state, world, obstacles, options)) {
+      return state;
+    }
+  }
+  if (!isStateValid(p2, world, obstacles, options)) {
+    return p2;
+  }
+  return std::nullopt;
+}
+
+PathPlanner::SplitResult PathPlanner::splitSegmentIfNecessary(
+  Path & path, const std::size_t ind1,
+  const std::size_t ind2, const World & world,
+  const std::vector<ateam_geometry::AnyShape> & obstacles,
+  const PlannerOptions & options)
+{
+  const auto maybe_collision_point = getCollisionPoint(
+    path.at(ind1), path.at(
+      ind2), world, obstacles, options);
+  if (!maybe_collision_point) {
+    return {
+      .split_needed = false,
+      .split_succeeded = true
+    };
+  }
+  const auto collision_point = maybe_collision_point.value();
+  const auto p1 = path[ind1];
+  const auto p2 = path[ind2];
+  auto change_vector = (ateam_geometry::normalize(p2 - p1) * kRobotRadius).perpendicular(
+    CGAL::CLOCKWISE);
+  auto new_point = collision_point;
+  auto direction = 1;
+  for (auto i = 1; i < 100; ++i) {
+    new_point = new_point + (change_vector * i * direction);
+    direction *= -1;
+    if (isStateValid(new_point, world, obstacles, options)) {
+      path.insert(path.begin() + ind2, new_point);
+      return {
+        .split_needed = true,
+        .split_succeeded = true
+      };
+    }
+  }
+  return {
+    .split_needed = true,
+    .split_succeeded = false
+  };
+}
+
+void PathPlanner::addRobotsToObstacles(
+  const World & world,
+  const ateam_geometry::Point & start_pos,
+  std::vector<ateam_geometry::AnyShape> & obstacles)
+{
+  auto obstacle_from_robot = [](const std::optional<Robot> & robot) {
+      return ateam_geometry::AnyShape(
+        ateam_geometry::makeCircle(robot.value().pos, kRobotRadius));
+    };
+
+  auto not_current_robot = [&start_pos](const std::optional<Robot> & robot) {
+      // Assume any robot close enough to the start pos is the robot trying to navigate
+      return (robot.value().pos - start_pos).squared_length() > std::pow(kRobotRadius, 2);
+    };
+
+  auto our_robot_obstacles = world.our_robots |
+    std::views::filter(std::mem_fn(&std::optional<Robot>::has_value)) |
+    std::views::filter(not_current_robot) |
+    std::views::transform(obstacle_from_robot);
+  obstacles.insert(
+    obstacles.end(),
+    our_robot_obstacles.begin(),
+    our_robot_obstacles.end());
+
+  auto their_robot_obstacles = world.their_robots |
+    std::views::filter(std::mem_fn(&std::optional<Robot>::has_value)) |
+    std::views::transform(obstacle_from_robot);
+  obstacles.insert(
+    obstacles.end(),
+    their_robot_obstacles.begin(),
+    their_robot_obstacles.end());
+}
+
+void PathPlanner::addDefaultObstacles(
```

Comment:
At some point, we want to move some of the common field geometries (goal box, center line, etc) to a separate file in ateam_common or something to make it easier to reference them